Index: app/src/main/java/com/example/jarsical/SongCollection.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/example/jarsical/SongCollection.java	(revision 3fd1c80e02021c9dbcdeda87dd33a6d1955c1dd2)
+++ app/src/main/java/com/example/jarsical/SongCollection.java	(date 1624845774736)
@@ -1,7 +1,7 @@
 package com.example.jarsical;
 
 public class SongCollection {
-    private Song songs[] = new Song[3];
+    private Song songs[] = new Song[2];
     public SongCollection(){
 
         Song photograph = new Song("S1001",
Index: app/src/main/java/com/example/jarsical/PlaySongActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/example/jarsical/PlaySongActivity.java	(revision 3fd1c80e02021c9dbcdeda87dd33a6d1955c1dd2)
+++ app/src/main/java/com/example/jarsical/PlaySongActivity.java	(date 1624845893176)
@@ -53,7 +53,7 @@
     }
     public void playSong(String songUrl){
         try{
-
+            player.reset();
             player.setDataSource(songUrl);
             player.prepare();
             player.start();
@@ -82,6 +82,7 @@
         Log.d("Temasek", "After playNext, the index is now: " + currentIndex);
         displaySongBasedOnIndex(currentIndex);
         playSong(fileLink);
+
     }
 
     public void playPrevious(View view){
@@ -91,6 +92,7 @@
         Log.d("Temasek", "After playNext, the index is now: " + currentIndex);
         displaySongBasedOnIndex(currentIndex);
         playSong(fileLink);
+
     }
 
     private void gracefullyStopsWhenMusicEnds(){
